### PR TITLE
Add support for Polaris 

### PR DESCRIPTION
**Pull request by ai03-2725**
## Description

Adds support for the Polaris keyboard.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Adds support for the Polaris keyboard

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/ai03/polaris/config.h [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+// /*== customize breathing effect ==*/
+//   /*==== (DEFAULT) use fixed table instead of exp() and sin() ====*/
+//   #define RGBLIGHT_BREATHE_TABLE_SIZE 256      // 256(default) or 128 or 64
+//   /*==== use exp() and sin() ====*/
+//   #define RGBLIGHT_EFFECT_BREATHE_CENTER 1.85  // 1 to 2.7
+//   #define RGBLIGHT_EFFECT_BREATHE_MAX    255   // 0 to 255
+#endif
+
+/* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
+#define DEBOUNCING_DELAY 5
```
Comment (drashna) [inline keyboards/ai03/polaris/config.h]:
```suggestion
#define DEBOUNCE 5
```

File: keyboards/ai03/polaris/readme.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Polaris
+
+![Polaris](https://i.imgur.com/bENjPiK.jpg)
+
+A basic 60% keyboard
+
+Keyboard Maintainer: [ai03](https://github.com/ai03-2725)
```
Comment (fauxpark) [inline keyboards/ai03/polaris/readme.md]:
```suggestion
* Keyboard Maintainer: [ai03](https://github.com/ai03-2725)
```

File: keyboards/ai03/polaris/readme.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# Polaris
+
+![Polaris](https://i.imgur.com/bENjPiK.jpg)
+
+A basic 60% keyboard
+
+Keyboard Maintainer: [ai03](https://github.com/ai03-2725)  
+Hardware Supported: Polaris PCB/case
```
Comment (fauxpark) [inline keyboards/ai03/polaris/readme.md]:
```suggestion
* Hardware Supported: Polaris PCB/case
```

File: keyboards/ai03/polaris/readme.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+Hardware Availability: Will be posted (here)[https://kb.ai03.me/projects/polaris.html]
```
Comment (fauxpark) [inline keyboards/ai03/polaris/readme.md]:
```suggestion
* Hardware Availability: Will be posted (here)[https://kb.ai03.me/projects/polaris.html]
```

File: keyboards/ai03/polaris/rules.mk [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+# If you don't know the bootloader type, then you can specify the
+# Boot Section Size in *bytes* by uncommenting out the OPT_DEFS line
+# Otherwise, delete this section
+#   Teensy halfKay      512
+#   Teensy++ halfKay    1024
+#   Atmel DFU loader    4096
+#   LUFA bootloader     4096
+#   USBaspLoader        2048
+# OPT_DEFS += -DBOOTLOADER_SIZE=4096
+
```
Comment (fauxpark) [inline keyboards/ai03/polaris/rules.mk]: Please remove this block.

Comment (noroadsleft) [inline keyboards/ai03/polaris/rules.mk]: Lines 14-23 should be deleted.

File: keyboards/ai03/polaris/readme.md (hunk 2 above)
Comment (noroadsleft) [inline keyboards/ai03/polaris/readme.md]: We prefer this section as a list now.

```suggestion
* Keyboard Maintainer: [ai03](https://github.com/ai03-2725)
```

File: keyboards/ai03/polaris/readme.md (hunk 3 above)
Comment (noroadsleft) [inline keyboards/ai03/polaris/readme.md]:
```suggestion
* Hardware Supported: Polaris PCB/case
```

File: keyboards/ai03/polaris/readme.md (hunk 4 above)
Comment (noroadsleft) [inline keyboards/ai03/polaris/readme.md]: Your bracket usage is backwards here.

```suggestion
* Hardware Availability: Will be posted [here](https://kb.ai03.me/projects/polaris.html)
```

File: keyboards/ai03/polaris/readme.md [hunk 6]
```diff
@@ -0,0 +1,15 @@
+# Polaris
+
+![Polaris](https://i.imgur.com/bENjPiK.jpg)
```
Comment (noroadsleft) [inline keyboards/ai03/polaris/readme.md]: Serves an image that's a bit smaller visually.

```suggestion
![Polaris](https://i.imgur.com/bENjPiKh.jpg)
```

File: keyboards/ai03/polaris/polaris.h [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+) \
+{ \
+    { K000,  K001,  K002,  K003,  K004,  K005,  K006,  K007,  K008,  K009,  K010,  K011,  K012,  K013 }, \
+	{ K100,  K101,  K102,  K103,  K104,  K105,  K106,  K107,  K108,  K109,  K110,  K111,  K112,  K113 }, \
+	{ K200,  K201,  K202,  K203,  K204,  K205,  K206,  K207,  K208,  K209,  K210,  K211,  K212,  K213 }, \
+	{ K300,  K301,  K302,  K303,  K304,  K305,  K306,  K307,  K308,  K309,  K310,  K311,  K312,  K313 }, \
+	{ K400,  K401,  K402,  KC_NO, K404,  KC_NO, K406,  KC_NO, K408,  KC_NO, K410,  K411,  K412,  K413 }  \
+}
+
+#define LAYOUT_ansi_tsangan( \
```
Comment (noroadsleft) [inline keyboards/ai03/polaris/polaris.h]: This is [`LAYOUT_60_ansi_tsangan` ](https://github.com/qmk/qmk_firmware/tree/c3200aa8fcf86122736846017ca7b125ac225842/layouts/default/60_ansi_tsangan/) with a split right shift.

```suggestion
#define LAYOUT_60_ansi_tsangan_split_rshift( \
```

File: keyboards/ai03/polaris/polaris.h [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+/* This a shortcut to help you visually see your layout.
+ *
+ * The first section contains all of the arguments representing the physical
+ * layout of the board and position of the keys.
+ *
+ * The second converts the arguments into a two-dimensional array which
+ * represents the switch matrix.
+ */
+#define LAYOUT_all( \
+    K000, K001, K002, K003, K004, K005, K006, K007, K008, K009, K010, K011, K012, K013, K212, \
```
Comment (noroadsleft) [inline keyboards/ai03/polaris/polaris.h]: Would you please clean up the indenting in this file? You're mixing tabs and spaces here. (Lines 29-40 and 44-55)

File: keyboards/ai03/polaris/polaris.c [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "polaris.h"
+
+// Optional override functions below.
+// You can leave any or all of these undefined.
+// These are only required if you want to perform custom actions.
+
```
Comment (noroadsleft) [inline keyboards/ai03/polaris/polaris.c]: Lines 18-21 can be deleted.

File: keyboards/ai03/polaris/keymaps/testing/readme.md [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+ESC = Backlight Toggle  
+1   = Backlight Step  
+  
+5   = RGB Toggle  
+6   = RGB Mode Cycle  
+7   = RGB Hue Increase  
+8   = RGB Hue Decrease  
+9   = RGB Saturation Increase  
+10  = RGB Saturation Decrease  
+11  = RGB Value Increase
```
Comment (noroadsleft) [inline keyboards/ai03/polaris/keymaps/testing/readme.md]:
```suggestion
MINUS  = RGB Value Increase  
```

File: keyboards/ai03/polaris/keymaps/testing/readme.md [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+1   = Backlight Step  
+  
+5   = RGB Toggle  
+6   = RGB Mode Cycle  
+7   = RGB Hue Increase  
+8   = RGB Hue Decrease  
+9   = RGB Saturation Increase  
+10  = RGB Saturation Decrease  
+11  = RGB Value Increase  
+12  = RGB Value Decrease
```
Comment (noroadsleft) [inline keyboards/ai03/polaris/keymaps/testing/readme.md]:
```suggestion
EQUALS  = RGB Value Decrease  
```

File: keyboards/ai03/polaris/keymaps/default_ansi_tsangan/keymap.c [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+ */
+#include QMK_KEYBOARD_H
+
+enum layer_names {
+    _BASE,
+    _FN
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [_BASE] = LAYOUT_ansi_tsangan( /* Base */
```
Comment (noroadsleft) [inline keyboards/ai03/polaris/keymaps/default_ansi_tsangan/keymap.c]:
```suggestion
  [_BASE] = LAYOUT_60_ansi_tsangan_split_rshift( /* Base */
```

File: keyboards/ai03/polaris/keymaps/default_ansi_tsangan/keymap.c [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+    KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_DEL,
+    KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC,          KC_BSLS,
+    MO(1),   KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,                   KC_ENT,
+    KC_LSFT,          KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,          KC_RSFT, KC_DEL,
+    KC_LCTL, KC_LGUI, KC_LALT,                                     KC_SPC,                                      KC_RALT, KC_RGUI, KC_RCTL
+  ),
+  [_FN] = LAYOUT_ansi_tsangan( /* FN */
```
Comment (noroadsleft) [inline keyboards/ai03/polaris/keymaps/default_ansi_tsangan/keymap.c]:
```suggestion
  [_FN] = LAYOUT_60_ansi_tsangan_split_rshift( /* FN */
```

File: keyboards/ai03/polaris/info.json [hunk 14]
```diff
@@ -0,0 +1,145 @@
...
+                {"x":3.75, "y":4, "w":2.25},
+                {"x":6, "y":4, "w":1.25},
+                {"x":7.25, "y":4, "w":2.75},
+                {"label":"Alt", "x":10, "y":4, "w":1.25},
+                {"label":"Win", "x":11.25, "y":4, "w":1.25},
+                {"label":"Menu", "x":12.5, "y":4, "w":1.25},
+                {"label":"Ctrl", "x":13.75, "y":4, "w":1.25}
+            ]
+        },
+        "LAYOUT_ansi_wkl": {
```
Comment (noroadsleft) [inline keyboards/ai03/polaris/info.json]:
```suggestion
        "LAYOUT_60_ansi_tsangan_split_rshift": {
```

File: keyboards/ai03/polaris/keymaps/testing/readme.md [hunk 15]
```diff
@@ -0,0 +1,17 @@
...
+  
+ESC = Backlight Toggle  
+1   = Backlight Step  
+  
+5    = RGB Toggle  
+6    = RGB Mode Cycle  
+7    = RGB Hue Increase  
+8    = RGB Hue Decrease  
+9    = RGB Saturation Increase  
+10   = RGB Saturation Decrease
```
Comment (fauxpark) [inline keyboards/ai03/polaris/keymaps/testing/readme.md]:
```suggestion
0    = RGB Saturation Decrease  
```

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**